src/services/SessionEventProcessor.ts

```diff
@@ -92,5 +115,26 @@ export class SessionEventProcessor {
 			throw new AppError(HttpCodesEnum.SERVER_ERROR, error.message);
 		}
 	}
+
+	async sendOldTemplateEmail(oldSessionEvent: SessionEvent): Promise<void> {
+		// Validate all necessary fields are populated before processing the data.
+		try {
+			await this.validationHelper.validateModel(oldSessionEvent, this.logger);
+		} catch (error) {
+			this.logger.error("Unable to process the DB record as the necessary fields are not populated to send the old template email.", { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS_IN_SESSION_EVENT });
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Unable to process the DB record as the necessary fields are not populated to send the old template email.");
+		}
+		// Send SQS message to GovNotify queue to send email to the user.
+		try {
+			const nameParts = personalIdentityUtils.getNames(oldSessionEvent.nameParts);
+			await this.iprService.sendToGovNotify(buildOldGovNotifyEventFields(oldSessionEvent.userId, oldSessionEvent.userEmail, nameParts.givenNames[0], nameParts.familyNames[0]));
+		} catch (error) {
+			this.logger.error("FAILED_TO_WRITE_GOV_NOTIFY", {
+				reason: "Processing Event session data, failed to post old email message to GovNotify SQS Queue",
+				error,
+			}, { messageCode: MessageCodes.FAILED_TO_WRITE_GOV_NOTIFY });
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "An error occurred when sending old email message to GovNotify handler");
```

Comment:
We are catching here and then throwing to be caught on lie 96 and then thrown again. Do we need both?

Reply:
I have refactored this file as much possible and should be readable now and easy to go through. Can you please review the latest changes and provide inputs if required.